python/S3_OBJECT_LOCK_ENABLED/S3_OBJECT_LOCK_ENABLED.py

```diff
@@ -0,0 +1,184 @@
+# Copyright 2017-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the License is located at
+#
+#        http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for
+# the specific language governing permissions and limitations under the License.
+'''
+#####################################
+##           Gherkin               ##
+#####################################
+Rule Name:
+  S3_OBJECT_LOCK_ENABLED
+
+Description:
+  Check whether S3 Buckets are Object Lock enabled. There should be a retention mode and retention
+  period to be applied to the bucket lock.
+
+Rationale:
+  Object Lock enabled on all S3 buckets to block object version deletion.
+
+Indicative Severity:
+  Medium
+
+Trigger:
+  Periodic
+
+Reports on:
+  AWS::S3::Bucket
+
+Rule Parameters:
+  Mode
+    The Object Lock retention mode you want to apply to new objects placed in the specified bucket
+  Days
+    The number of days that you want to specify for the default retention period.
+  Years
+    The number of years that you want to specify for the default retention period.
+
+Scenarios:
+  Scenario: 1
+    Given: No S3 buckets available for the account
+     Then: Return NOT_APPLICABLE
+  Scenario: 2
+    Given: S3 bucket is not object lock enabled
+     Then: Return NON_COMPLIANT
+  Scenario: 3
+    Given: S3 bucket is object lock enabled with mode
+     Then: Return COMPLIANT
+  Scenario: 4
+    Given: S3 bucket is object lock enabled with mode and days
+          (actual bucket days should be greater than the parameter days)
+     Then: Return COMPLIANT
+  Scenario: 5
+    Given: S3 bucket is object lock enabled with mode and years
+          (actual bucket years should be greater than the parameter years)
+     Then: Return COMPLIANT
+  Scenario: 6
+    Given: S3 bucket is object lock enabled with mode, days and yeaas
+          (actual bucket days(days + (years*365)) should be greater than the parameter days
+     Then: Return COMPLIANT
+  Scenario: 7
+    Given: S3 bucket is object lock enabled with different mode
+     Then: Return NON_COMPLIANT
+  Scenario: 8
+    Given: S3 bucket is object lock enabled with mode and days
+          (actual bucket days are less than the parameter days)
```

Comment:
use And condition for condition check to decide Complaint or Non-Complaint- Given And Then